### ISSUE TITLE
RocketCDN auto-resume clears the entire cache on every admin_init when the subscription is cancelled (3.22.0.2 regression)

### ISSUE THREAD

**Issue by sebabal** [wp-media/wp-rocket#8555]
## Description

Since **3.22.0.2**, `WP_Rocket\Engine\CDN\Render\Controller::maybe_auto_create_rocketcdn_free_subscription()` is hooked on **`admin_init`** and calls `$this->cache->clear_all_cache()` (i.e. `rocket_clean_domain()` — a **full page-cache purge**) on **every invocation** when the RocketCDN subscription is `cancelled`/`refunded` and the site is outside the grace period.

`admin_init` fires on every `wp-admin` request **and every `admin-ajax.php` request** (Action Scheduler, Heartbeat, WooCommerce, cron-over-ajax, etc.). On any site with routine background admin-ajax traffic this fires many times per minute, so the **entire page cache is wiped continuously and never persists** — every front-end request becomes a full uncached render.

There is no idempotency guard: the gating condition (`is_cancelled_outside_grace_period()`) is a persistent *state*, not a one-time *event*, so the cache clear repeats on every `admin_init` indefinitely.

## Affected version

Introduced in **3.22.0.2** (the auto-resume / "subscription state handling and transitions" change). The methods involved are tagged `@since 3.22.0.2`.

## Environment

- WP Rocket: **3.22.0.2**
- WordPress: 7.0
- PHP: 8.3.6
- Object cache: Redis (`redis-cache`)
- Other: WPML (multilingual, large URL space), nginx, Preload + Remove Unused CSS enabled (Action-Scheduler-driven admin-ajax traffic)
- RocketCDN: Free tier, subscription status = **cancelled** (CDN CNAMEs empty)

## Steps to reproduce

1. Be on WP Rocket 3.22.0.2 with a RocketCDN subscription in a `cancelled`/`refunded` state, site not pending deletion (`is_cancelled_outside_grace_period()` returns `true`).
2. Generate any routine `admin-ajax.php` traffic (e.g. Action Scheduler processing a queue, Heartbeat, or load `/wp-admin/` repeatedly).
3. Watch `wp-content/cache/wp-rocket/<host>/` — the directory is deleted and recreated repeatedly; cached pages never survive.

## Expected behaviour

A cancelled RocketCDN subscription should not cause the WP Rocket **page cache** to be cleared on normal admin requests. Any cache clear tied to a subscription/CDN state change should happen once per actual state transition, not on every `admin_init`.

## Actual behaviour

The full page cache is purged on every `admin_init`. With routine background admin-ajax activity this happens continuously, so the cache is effectively always empty and every request is a cold render.

## Root cause

`inc/Engine/CDN/Render/Controller.php` (~lines 663–690):

```php
public function maybe_auto_create_rocketcdn_free_subscription() {
    if ( ! $this->subscription_controller->is_cancelled_outside_grace_period() ) {
        return;
    }
    if ( $this->subscription_controller->is_paid() && $this->subscription_controller->is_in_grace_period() ) {
        return;
    }
    if ( $this->subscription_controller->is_license_invalid() ) {
        return;
    }

    $stored               = $this->get_forced_pause_tracking();
    $stored['persistent'] = false;
    update_option( self::FORCED_PAUSE_TRACKING_OPTION, $stored, false );

    // Clear whole cache.
    $this->cache->clear_all_cache();           // <-- runs on EVERY admin_init while cancelled

    if ( empty( $this->get_items() ) ) {
        return;
    }
    $this->subscription_controller->create_subscription();
}
```

`inc/Engine/CDN/RocketCDN/SubscriptionController.php`:

```php
public function is_cancelled_outside_grace_period(): bool {
    return $this->is_cancelled() && ! $this->is_website_pending_deletion();
}

public function is_cancelled(): bool {
    $subscription = $this->get_subscription_data();
    return ! empty( $subscription['subscription_status'] )
        && in_array( $subscription['subscription_status'], [ 'cancelled', 'refunded' ], true );
}
```

`inc/Engine/CDN/Cache.php`:

```php
public function clear_all_cache(): void {
    rocket_clean_domain();   // full domain page-cache purge
}
```

`maybe_auto_create_rocketcdn_free_subscription` is registered on `admin_init` via `WP_Rocket\Engine\CDN\Render\Subscriber`. The function proceeds whenever the subscription is cancelled (a stable state, true on every request) and unconditionally calls `clear_all_cache()` **before** the `get_items()` early-return, with no flag recording that the clear already happened — so it clears on every `admin_init`.

## Confirmed backtrace (captured on `before_rocket_clean_domain`)

```
do_action('admin_init')
  → WP_Rocket\Engine\CDN\Render\Subscriber->maybe_auto_create_rocketcdn_free_subscription
  → WP_Rocket\Engine\CDN\Render\Controller->maybe_auto_create_rocketcdn_free_subscription
  → WP_Rocket\Engine\CDN\Cache->clear_all_cache
  → rocket_clean_domain
  → do_action('before_rocket_clean_domain')
```

This fires repeatedly (on every `admin_init`); cached pages never persist beyond a few seconds.

**Comment by punto-rf** [comment]
I have the same issue.

After updating to 3.22.0.2, my cache started behaving abnormally.

The contents of wp-content/cache/wp-rocket/ were constantly being replaced instead of growing normally, and the preload queue got stuck with rocket_preload_job_preload_url remaining in progress.

Rolling back to 3.21.3 immediately fixed the issue and the cache became stable again.

**Comment by hf-pablop** [comment]
I have the same issue.

**Comment by suzoutlet** [comment]
Closing this. 
This has now been fixed in [3.22.0.3 ](https://wp-rocket.me/changelog/)
